six/modules/python/cphd/source/cphd.i

```diff
@@ -143,6 +144,289 @@ using six::Vector3;
+                    length = dtype.itemsize * num_values
+
+                    # For every vector (first index of channel_data), slice
+                    # the range with values for this parameter (second index)
+                    vals = channel_data[
+                        :, pvp_offset:(pvp_offset + length)
+                    ].copy().view(dtype)
+                    channel_pvp[_name] = vals.reshape(
+                        num_vectors, num_values
+                    ).squeeze()
+                    pvp_offset += 8 * _size  # Move to next 8-byte word
+
+            pvp_data.append(channel_pvp)
+
+        return pvp_data
+
+    @staticmethod
+    def from_list_of_dicts(pvp_data, metadata):
+        """Initialize and populate a PVPBlock from a list of Python dicts.
+
+        Parameters
+        ----------
+        pvp_data: list of Python dicts
+            List of Python dicts (one for each channel) mapping parameter names
+            to NumPy arrays of data. See PVPBlock.toListOfDicts() for more info
+            on the structure expected here
+        metadata: cphd.Metadata
+            Metadata used to create this PVPBlock
+        """
+        pvp_block = PVPBlock(metadata.pvp, metadata.data)
+
+        # Methods to check if an optional PVP param is set
+        # Using .lower() since name capitalization is not 100% consistent
+        has_optional_param_methods = {name.lower(): getattr(pvp_block, name)
+                                      for name in dir(pvp_block)
+                                      if name.startswith('has')}
+
+        pvp_info = {}
+
+        # Set required and optional PVP parameters
+        for name in dir(metadata.pvp):
+            try:
+                attr = getattr(metadata.pvp, name)
+            except AttributeError:
+                continue
+            if not isinstance(attr, pysix.cphd.PVPType):
+                continue
+            if attr.getSize() == 0:
+                continue
+            # Is this an optional parameter that is not set?
+            if ('has' + name.lower()) in has_optional_param_methods \
+                    and not has_optional_param_methods['has' + name.lower()]():
+                continue
+            pvp_info[name] = (
+                attr.getSize(), attr.getOffset(), attr.getFormat()
+            )
+
+        # Using .lower() since name capitalization is not 100% consistent
+        default_params_in_use = [name.lower() for name in pvp_info]
+        default_param_setters = {
+            name[3:].lower(): getattr(pvp_block, name)
+            for name in dir(pvp_block)
+            if name.startswith('set')
+            and name[3:].lower() in default_params_in_use
+        }
+
+        # Populate PVPBlock object from pvp_data
+        for channel_index, channel_data in enumerate(pvp_data):
+            for vector_index in range(metadata.getNumVectors(channel_index)):
+                for name, data in channel_data.items():
```

Comment:
It might be good to check that the list of keys in `channel_data` matches the list of PVP params (the required ones, any optional ones, and any added ones)